api/desecapi/pdns.py

```diff
@@ -0,0 +1,104 @@
+import requests
+import json
+from desecapi import settings
+
+
+headers = {
+    'User-Agent': 'desecapi',
+    'X-API-Key': settings.POWERDNS_API_TOKEN,
+}
+
+
+def normalize_hostname(name):
+    if '/' in name or '?' in name:
+        raise Exception('Invalid hostname ' + name)
+    return name if name.endswith('.') else name + '.'
+
+
+def _pdns_post(url, body):
+    r = requests.post(settings.POWERDNS_API + url, data=json.dumps(body), headers=headers)
+    if r.status_code < 200 or r.status_code >= 300:
+        raise Exception(r.text)
+    return r
+
+
+def _pdns_patch(url, body):
+    r = requests.patch(settings.POWERDNS_API + url, data=json.dumps(body), headers=headers)
+    if r.status_code < 200 or r.status_code >= 300:
+        raise Exception(r.text)
+    return r
+
+
+def _pdns_get(url):
+    r = requests.get(settings.POWERDNS_API + url, headers=headers)
+    if (r.status_code < 200 or r.status_code >= 300) and r.status_code != 404:
+        raise Exception(r.text)
+    return r
+
+
+def _delete_or_replace(name, type, value):
+    """
+    Return pdns API json to either replace or delete a record, depending on value is empty or not.
```

Comment:
I changed the function name to `_delete_or_replace_rrset` and updated the doc comment